custom_components/sector/coordinator.py

```diff
@@ -60,8 +91,23 @@ async def _async_update_data(self):
                     data[key] = value
                     data[key]["code_format"] = self.code_format
 
+            # Retrieve and filter logs
+            raw_logs = api_data.get("Logs", [])
+            logs = []
+
+            # Ensure raw_logs is a list; handle unexpected formats
+            if isinstance(raw_logs, list):
+                logs = self._filter_duplicate_logs(raw_logs)
+            elif isinstance(raw_logs, dict) and "Records" in raw_logs:
+                logs = self._filter_duplicate_logs(raw_logs.get("Records", []))
+
+            # Update last_processed_events to track processed logs
+            self.last_processed_events.update(
```

Comment:
There is no need to manage a queue, just have the last 40 events on hand.

Reply:
Queue nuked.